### ISSUE TITLE
Check if `new BigInteger("0") != zero` works

### ISSUE THREAD

**Issue by matcauthon49** [TypeCrypt/OasisGKR#2]

**Comment by ShouvikGhosh2048** [comment]
```
import java.math.BigInteger;

class Main {
    public static void main(String [] args) {
        BigInteger zero = new BigInteger("0");
        BigInteger x = new BigInteger("0");
        BigInteger y = new BigInteger("1");
        System.out.println(x == zero);
        System.out.println(x.equals(zero));
        System.out.println(y == zero);
        System.out.println(y.equals(zero));
    }
}
```
prints

> false
> true
> false
> false

**Comment by matcauthon49** [comment]
Fixed with commit 01354d2f5c89af648610f84c63ad4d4e8c77cc2e.
